llvm/utils/repo/repo.cmake

```diff
@@ -24,10 +24,57 @@ else ()
     set (utils_dir /usr/share/repo)
 endif ()
 
-SET (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -fno-exceptions -fno-rtti" CACHE STRING "Default C Flags Debug")
-SET (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -fno-exceptions -fno-rtti" CACHE STRING "Default CXX Flags Debug")
-SET (CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -fno-exceptions -fno-rtti" CACHE STRING "Default C Flags Release")
-SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -fno-exceptions -fno-rtti" CACHE STRING "Default CXX Flags Release")
+# The user may specify the libcxxabi_include by giving
+# '-Dlibcxxabi_include:STRING=/path/to/clang/include' on the command line
+if (libcxxabi_include)
+	set (libcxxabi_include "${libcxxabi_include}")
+endif ()
+
+# The user may specify the exception and rtti by giving
+# '-Dexception_rtti:BOOL=Yes' on the command line
+if (exceptions_rtti)
+	set (exceptions_rtti "-fexceptions -frtti")
+else ()
+	set (exceptions_rtti "-fno-exceptions -fno-rtti")
+endif ()
+
+# The user may specify the musl_install by giving
+# '-Dmusl_install:STRING=/path/to/musl/installed/directory' on the command line
+if (musl_install)
+	set (musl_install "${musl_install}")
+else ()
+	set (musl_install "/home/prepo/musl")
+endif ()
+
+# The user may specify the llvm_install by giving
+# '-Dllvm_install:STRING=/path/to/llvm/installed/directory' on the command line
+if (llvm_install)
+	set (llvm_install "${llvm_install}")
+else ()
+	set (llvm_install "/home/prepo/LLVM")
+endif ()
+
+# The user may use the LLVM libc++.a by giving
+# '-Dlibcxx:BOOL=Yes' on the command line
+if (libcxx)
+	set (libcxx "-stdlib=libc++")
+	set (libcxxabi_lib "-stdlib=libc++ -lc++abi")
+endif()
+
+# The user may use the MUSL libc.a by giving
+# '-Dmusl:BOOL=Yes' on the command line
+if (musl)
+	SET (MUSL_COMPILE_FLAGS "-nostdinc -nostdinc++ --sysroot ${musl_install} -isystem ${musl_install}/include ${libcxxabi_include}")
+	SET (CMAKE_EXE_LINKER_FLAGS "-nostdlib -nodefaultlibs -static --sysroot ${musl_install} -L ${musl_install}/lib -L ${llvm_install}/lib ${libcxxabi_lib} -lc" CACHE STRING "toolchain_exelinkflags" FORCE)
```

Comment:
Is there a deliberate choice in setting this variable rather than the existing  CMAKE_XX_LINK_EXECUTABLE variables below?

Reply:
`CMAKE_EXE_LINKER_FLAGS` is a Linker flag to be used to create executables. It is used by the `CMAKE_XX_LINK_EXECUTABLE` variables below. `CMAKE_EXE_LINKER_FLAGS` contains the linker options which are used by both `CMAKE_C_LINK_EXECUTABLE` and `CMAKE_CXX_LINK_EXECUTABLE`.